server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -993,8 +999,23 @@ public void deleteSnapshot(final String repositoryName, final String snapshotNam
 
             @Override
             public ClusterState execute(ClusterState currentState) {
+                if (snapshotNames.size() > 1 && currentState.nodes().getMinNodeVersion().before(MULTI_DELETE_VERSION)) {
+                    throw new IllegalArgumentException("Deleting multiple snapshots in a single request is only supported in version [ "
+                            + MULTI_DELETE_VERSION + "] but cluster contained node of version [" + currentState.nodes().getMinNodeVersion()
+                            + "]");
+                }
                 final SnapshotsInProgress snapshots = currentState.custom(SnapshotsInProgress.TYPE);
-                final SnapshotsInProgress.Entry snapshotEntry = findInProgressSnapshot(snapshots, snapshotName, repositoryName);
+                final SnapshotsInProgress.Entry snapshotEntry;
+                if (snapshotNames.size() == 1) {
+                    final String snapshotName = snapshotNames.iterator().next();
+                    if (Regex.isSimpleMatchPattern(snapshotName)) {
+                        snapshotEntry = null;
+                    } else {
+                        snapshotEntry = findInProgressSnapshot(snapshots, snapshotName, repositoryName);
+                    }
+                } else {
+                    snapshotEntry = null;
```

Comment:
do we need to throw a Concurrent...Exception in case where there is `SnapshotsInProgress.Entry` but we did not look at aborting it?

Reply:
We do that check in `deleteCompletedSnapshots` so we don't need it here.
Technically we could add it here as well, but now that we cache `RepositoryData` it wouldn't do much to duplicate it here (we have to run it again before putting the delete entry into the CS anyway) except for saving a noop CS update task in case of a concurrent snapshot. -> I figured I'd not duplicate logic.